### ISSUE TITLE
[Feature request] Small improvements for session and templates saving

### ISSUE THREAD

**Issue by zigmhount** [Houston4444/RaySession#221]
Hi,
I use RaySession all the time, thanks a bunch for your work :)

However I find tiny annoyances in my personal workflow for saving sessions and templates, and I think it may be quite straightforward to implement these small improvements:
1. When saving a template, I'd like to have a list of the existing templates, with the possibility of selecting one from the list to overwrite it. Typically, I use a template for a new project, and then implement improvements that I want to save in the template, but currently I can only overwrite the existing template if I'm lucky to remember the existing template's exact name.
2. When doing step 1, ask for confirmation to overwrite the existing template.
3. Add an option "Save As" in the Session menu. Typically, sometimes I modify an existing session which over time becomes something different, even though I had not thought about it beforehand (otherwise I would have first duplicated it), so I'd prefer to save the current session with a new name (and the one I initially opened is not modified). This may make it complicated to handle with the tracking via git?

What do you think?

**Comment by Houston4444** [comment]
Hi !
> I use RaySession all the time, thanks a bunch for your work :)
Nice to hear, thanks for this.

1 and 2 are perfectly understandable, possible, and not so long to implement, I think it is a good idea, it could be done in the future (not near future).
Something I am not sure to understand for the 3 is
* do you want a 'Duplicate' without saving current session ?
  * in this case, I don't want to name it 'save as', because user waits a quick action, and it won't be the case.
  * a checkbox 'save the current session' in the duplicate dialog would do the job (as it exists in 'open session' dialog'), and on server side it is very easy.
* or do you want 'Save as' but keep the current session running ?
  * it's possible, it just have to copy the session directory and do the rename stuff inside
* If you want a fast 'save as' and switch session, it is structurally impossible to warranty

**Comment by Houston4444** [comment]
Woow, forget what I said, I spoke too fastly, 'duplicate' without saving current session, would duplicate the session without pending changes. To work correctly, on server side it would mean:
* copy the current session folder elsewhere (we can't simply move it because of non running clients)
* save the session
* rename the session
* switch to the renamed session
* move the copied session folder to its original place

So, it is not so simple, but doable. Sorry, I speak alone, but it will give me the way to do if I attempt to do it (Not soon, I am on patchbay things).

**Comment by zigmhount** [comment]
Yeah, that's what I thought, but your outline is helpful to me as well to understand how the server works :)

As a comment though:
> I don't want to name it 'save as', because user waits a quick action
> [...]
> a checkbox 'save the current session' in the duplicate dialog would do the job (as it exists in 'open session' dialog')

I beg to differ. As a user, a "save as" menu entry is very standard and does not promise a quick action, on the contrary the user clicks this when he specifically want to set a name and destination to save into. Your checkbox idea doesn't seem intuitive to me, because the "Duplicate" button to first click on suggests that it clones something which is already saved (so I wouldn't click on "Duplicate" if I didn't want to save first, and I would be surprised to see this checkbox).
But it's probably only a matter of wording and documentation  - and potentially tooltips to make it crystal clear :)

In any case:
> 1 and 2 are perfectly understandable, possible, and not so long to implement, I think it is a good idea, it could be done in the future (not near future).

Great! No rush, I can definitely live with it as it is :)
